### PR TITLE
docs: clean up and clarify theming docs

### LINKED ISSUE #13206: Theming documents confusion or possible conflict
#### Bug, feature request, or proposal:
Proposal

#### What is the expected behavior?
Docs would not conflict or be more clear

#### What is the current behavior?

In the [Theming Guide - Custom Theme](https://material.angular.io/guide/theming#defining-a-custom-theme)

There is a paragraph at the end of that heading that reads...

> The theme file should not be imported into other SCSS files. This will cause duplicate styles to be written into your CSS output. If you want to consume the theme definition object (e.g., $candy-app-theme) in other SCSS files, then the definition of the theme object should be broken into its own file, separate from the inclusion of the mat-core and angular-material-theme mixins.

And then in another document regarding [theming your own components](https://material.angular.io/guide/theming-your-components#using-colors-from-a-palette) 

There is a code snippet that shows the theme file being imported into a component.scss file. So either this contradicts the previously mentioned docs or the docs should show us an example of how the definition of the theme object can be broken into its own file and make that obvious in that example.

#### What are the steps to reproduce?
read [Theming Guide - Custom Theme](https://material.angular.io/guide/theming#defining-a-custom-theme) then read [theming your own components](https://material.angular.io/guide/theming-your-components#using-colors-from-a-palette) 

#### What is the use-case or motivation for changing an existing behavior?
clarity?

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
v6

#### Is there anything else we should know?
I like turtles

### PR DESCRIPTION
**Pull request by jelbourn**
This makes some minor wording changes to the main theming guide and mostly rewrites the theming-your-own-component guide. Edited for clarify, brevity, and language.

Fixes #13206

### CODE REVIEW

File: guides/theming-your-components.md [hunk 1]
```diff
@@ -1,80 +1,81 @@
-# Theming your custom components
-In order to style your own components with Angular Material's tooling, the component's styles must be defined with Sass.
+### Theming your own component the Angular Material way
```
Comment (josephperrott) [inline guides/theming-your-components.md]: Should this be something like

> ### Theming your component using Angular Material's theming system

File: guides/theming-your-components.md [hunk 2]
```diff
@@ -1,80 +1,81 @@
...
-
-#### Why using `@mixin`
-The advantage of using a `@mixin` function is that when you change your theme, every file that uses it will be automatically updated.
-Calling the `@mixin` with a different theme argument allows multiple themes within the app or component.
-
-#### How to use `@mixin`
-We can better theme our custom components by adding a `@mixin` function to its theme file, and then call this function to apply a theme.
-
-All you need is to create a `@mixin` function in the custom-component-theme.scss
+#### 1. Define all color and typography styles in a "theme mixin" for the component
```
Comment (josephperrott) [inline guides/theming-your-components.md]: This section should include examples for both theme and typography, currently we mention typography but don't show it in the examples.

File: guides/theming-your-components.md (hunk 1 above)
Comment (jelbourn) [reply guides/theming-your-components.md]: Done

File: guides/theming-your-components.md (hunk 2 above)
Comment (jelbourn) [reply guides/theming-your-components.md]: Done

**Pull request merged**